k4FWCore/include/k4FWCore/BaseClass.h

```diff
@@ -0,0 +1,19 @@
+#ifndef K4FWCORE_FUNCTIONALUTILS_H
+#define K4FWCORE_FUNCTIONALUTILS_H
+
+#include "GaudiAlg/GaudiAlgorithm.h"
+#include "GaudiKernel/DataObjectHandle.h"
+#include "k4FWCore/DataWrapper.h"
+
+// Base class used for the Traits template argument of the
+// Gaudi::Functional algorithms
+struct BaseClass_t {
```

Comment:
This is a badly chosen name - and not even name spaced. Do you have something more prescriptive what the purpose is?

Reply:
Here the idea is that the Gaudi examples always use a `BaseClass_t` that is defined to be typically `Gaudi::Algorithm` so by having this the changes are minimal. See for example https://gitlab.cern.ch/gaudi/Gaudi/-/blob/master/GaudiExamples/src/FunctionalAlgorithms/MakeAndConsume.cpp#L60 and a few lines below; In k4FWCore we don't need line 60 but we need instead `#include "BaseClass.h"` and the rest is exactly the same as in the Gaudi examples. We could also name it differently and have it namespaced but I thought for consistency it would be simpler this way.